src/ray/core_worker/core_worker.cc

```diff
@@ -1429,56 +1440,73 @@ Status CoreWorker::SealExisting(const ObjectID &object_id,
 
 Status CoreWorker::ExperimentalChannelReadRelease(
     const std::vector<ObjectID> &object_ids) {
-  RAY_CHECK(object_ids.size() == 1);
-  return experimental_mutable_object_manager_->ReadRelease(object_ids[0]);
+  RAY_CHECK_EQ(object_ids.size(), 1UL);
+  return experimental_mutable_object_provider_->object_manager().ReadRelease(
+      object_ids[0]);
 }
 
-Status CoreWorker::ExperimentalChannelRegisterReader(const ObjectID &object_id) {
-  return ExperimentalChannelRegisterWriterOrReader(object_id, /*is_writer=*/false);
+Status CoreWorker::ExperimentalRegisterMutableObjectWriter(const ObjectID &object_id) {
+  std::unique_ptr<plasma::MutableObject> object;
+  RAY_RETURN_NOT_OK(
+      plasma_store_provider_->GetExperimentalMutableObject(object_id, &object));
+  RAY_CHECK(object) << "Mutable object must be local to be registered";
+  return experimental_mutable_object_provider_->object_manager().RegisterChannel(
```

Comment:
Perhaps we should move this code into MutableObjectProvider but only call it if the provided node_id matches ours?